tests/integration/conftest.py

```diff
@@ -154,28 +167,23 @@ async def isolated_subnet(client, subnet_resource):
         client.get(Pod, name="allowed-pod", namespace="allowed"),
     ]
 
-    pods = []
-    log.info("Waiting for pods...")
-    for pod in watch:
-        client.wait(
-            Pod,
-            pod.metadata.name,
-            for_conditions=["Ready"],
-            namespace=pod.metadata.namespace,
-        )
-        pods.append(await wait_pod_ip(client, pod))
+    pods = await wait_pod_ips(client, watch)
 
     yield tuple(pods)
 
     log.info("Deleting isolated subnet resources ...")
     for obj in codecs.load_all_yaml(path.read_text()):
         client.delete(type(obj), obj.metadata.name, namespace=obj.metadata.namespace)
 
+    await wait_for_removal(client, pods)
+
+
+async def wait_for_removal(client, pods):
```

Comment:
how about some docstring here to help indicate what this method does

```suggestion
async def wait_for_removal(client, pods):
    """Waits until listed pods are no longer present in the cluster."""
```